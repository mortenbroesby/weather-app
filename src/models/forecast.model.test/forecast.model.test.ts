import { ForecastModel } from "../forecast.model";

describe("models/forecast.model.ts", () => {
  it("Generates a valid Forecast model", () => {
    const SampleJson = require("./forecast.example.valid");
    const model = new ForecastModel(SampleJson);

    expect(model.itemCount).toBe(3);

    expect(model).toMatchSnapshot();
  });

  it("Generates a valid Forecast model without items", () => {
    const SampleJson = require("./forecast.example.valid.poorData");
    const model = new ForecastModel(SampleJson);

    expect(model).toMatchSnapshot();
  });

  it("Generates a valid Forecast model on malformed data", () => {
    const model = new ForecastModel();

    expect(model).toMatchSnapshot();
  });
});
