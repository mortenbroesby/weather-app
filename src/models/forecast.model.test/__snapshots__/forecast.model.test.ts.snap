// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`models/forecast.model.ts Generates a valid Forecast model 1`] = `
ForecastModel {
  "itemCount": 3,
  "items": Array [
    WeatherModel {
      "humidity": 70,
      "precipitation": 0,
      "temperature": Object {
        "current": 20,
        "maximum": 20,
        "minimum": 18,
      },
      "timestamp": 1566226800,
      "weatherType": Object {
        "description": "clear sky",
        "id": 800,
        "type": "Clear",
      },
      "windSpeed": 9,
    },
    WeatherModel {
      "humidity": 72,
      "precipitation": 0.688,
      "temperature": Object {
        "current": 20,
        "maximum": 20,
        "minimum": 18,
      },
      "timestamp": 1566237600,
      "weatherType": Object {
        "description": "light rain",
        "id": 500,
        "type": "Rain",
      },
      "windSpeed": 9,
    },
    WeatherModel {
      "humidity": 72,
      "precipitation": 1.25,
      "temperature": Object {
        "current": 19,
        "maximum": 19,
        "minimum": 18,
      },
      "timestamp": 1566248400,
      "weatherType": Object {
        "description": "light rain",
        "id": 500,
        "type": "Rain",
      },
      "windSpeed": 10,
    },
  ],
}
`;

exports[`models/forecast.model.ts Generates a valid Forecast model on malformed data 1`] = `
ForecastModel {
  "itemCount": 0,
  "items": Array [],
}
`;

exports[`models/forecast.model.ts Generates a valid Forecast model without items 1`] = `
ForecastModel {
  "itemCount": 0,
  "items": Array [],
}
`;
